packages/react-refresh/src/__tests__/ReactFreshIntegration-test.js

```diff
@@ -30,17 +33,21 @@ describe('ReactFreshIntegration', () => {
       React = require('react');
       ReactFreshRuntime = require('react-refresh/runtime');
       ReactFreshRuntime.injectIntoGlobalHook(global);
-      ReactDOM = require('react-dom');
-      act = React.unstable_act;
+      ReactDOMClient = require('react-dom/client');
+      Scheduler = require('scheduler/unstable_mock');
+      ({act, assertLog} = require('internal-test-utils'));
       container = document.createElement('div');
       document.body.appendChild(container);
+      root = ReactDOMClient.createRoot(container);
       exportsObj = undefined;
     }
   });
 
-  afterEach(() => {
+  afterEach(async () => {
     if (__DEV__) {
-      ReactDOM.unmountComponentAtNode(container);
+      await act(async () => {
+        root.unmount();
+      });
```

Comment:
Shouldn't need act here since unmount is wrapped in flushSync

Reply:
Thanks, I forgot that unmount is sync.